platforms/software/dependency-management/src/main/java/org/gradle/internal/component/resolution/failure/formatting/CapabilitiesDescriber.java

```diff
@@ -51,4 +53,12 @@ private static String describeCapability(Capability c) {
         }
         return '\'' + c.getGroup() + ":" + c.getName() + '\'';
     }
+
+    public static String describeCapabilitySelectorsWithTitle(Set<CapabilitySelector> capabilities) {
+        if (capabilities.size() == 1) {
+            return capabilities.iterator().next().getDisplayName();
+        }
+
+        return "[" + capabilities.stream().map(CapabilitySelector::getDisplayName).collect(Collectors.joining(", ")) + "]";
```

Comment:
💭 A little odd that this is the only `describeCapabilityX` method that uses `getDisplayName` instead of `describeCapability`.  Maybe a different name for this method would be better?

Reply:
This method describes capability selectors, not capabilities. The method name already denotes that

Reply:
I think the misalignment is in how the `getDisplayName` method is misimplemented.

`"capability selector for feature '" + featureName + "'"` isn't really a _display name_.  The name of this method says _describe_, maybe the selectors (which aren't named entities) should instead offer a `describe` method.  Then everything would look fine.

Reply:
The method `getDisplayName` comes from the interface `Describable`. Perhaps `Describable` should instead define a `getDescription` name instead, but we're way past changing that.